src/main/java/core/basesyntax/service/impl/DataConverterImpl.java

```diff
@@ -0,0 +1,33 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.model.Operation;
+import core.basesyntax.service.DataConverter;
+import java.util.ArrayList;
+import java.util.List;
+
+public class DataConverterImpl implements DataConverter {
+    @Override
+    public List<FruitTransaction> convertToTransaction(List<String> inputData) {
+        if (inputData.isEmpty()) {
+            return new ArrayList<>();
+        }
+
+        List<FruitTransaction> fruitTransactions = new ArrayList<>();
+        for (String line : inputData.subList(1, inputData.size())) {
+            String[] partsOfData = line.split(",");
+            if (partsOfData.length != 3) {
+                throw new IllegalArgumentException("Incorrect input data in line: " + line);
+            }
+            Operation fruitOperation = Operation.valueOf(partsOfData[0]);
```

Comment:
Using `valueOf` directly on the `Operation` enum can throw an `IllegalArgumentException` if the provided operation does not match any enum constant. You should handle this case and throw a more informative exception, or use a different approach that validates the operation before trying to convert it.